### ISSUE TITLE
Signup: tapping back after choosing to create/add a site does not go back to post-signup interstitial.

### ISSUE THREAD

**Issue by thehenrybyrd** [wordpress-mobile/WordPress-Android#15392]
### Expected behavior
I expect the back button to take me to the screen I was just on.

### Actual behavior
After creating a brand new WordPress.com account, the post-signup interstitial presents the option to "create WordPress.com site", "add a self-hosted site" or "not right now". Tapping "create WordPress.com site" brings us into the site creation flow, however tapping back at this point does not bring back the post-signup interstitial. 
Instead, we are brought to My Sites, which is a new screen for the user.

### Steps to reproduce the behavior
1. From a fully logged-out state, create a brand new WordPress.com account on the app.
2. Complete the signup process (email magic link) and note you wind up back in the app on the post-signup interstitial. 
3. Select an option.
4. Now, tap the back button
5. Note, you are not back on the post-signup interstitial you just came from. 

https://user-images.githubusercontent.com/15107387/135403652-b9c37ecc-61d8-4768-961c-d173243aa4f3.mov

##### Tested on Samsung Galaxy S21, Android 11, WPAndroid 18.3-rc-2

**Comment by dcalhoun** [comment]
I spent a little looking into this. I did not produce a solution, but will share what I observed for future work.

In the sign up epilogue, we [trigger the post sign up interstitial](https://github.com/wordpress-mobile/WordPress-Android/blob/42527b522ea82a79fefd974f2a9d5fa57eb8c520/WordPress/src/main/java/org/wordpress/android/ui/accounts/SignupEpilogueActivity.java#L58-L60), defined below.

https://github.com/wordpress-mobile/WordPress-Android/blob/42527b522ea82a79fefd974f2a9d5fa57eb8c520/WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java#L1397-L1403

This relies upon a `TaskStackBuilder`, which, from reading [its documentation](https://developer.android.com/reference/androidx/core/app/TaskStackBuilder), may imply using the back button to go back to this post sign up interstitial may not work by default. 

>The back key's behavior is local to the current task and does not capture navigation across different tasks. Navigating across tasks and easily reaching the previous task is accomplished through the "recents" UI, accessible through the software-provided Recents key on the navigation or system bar.

We may need to determine a way to conditionally reinitialize the post sign up interstitial when the users navigates backwards, but did not press the "Not right now" option found in the post sign up epilogue.

**Comment by jd-alexander** [comment]
> This relies upon a TaskStackBuilder, which, from reading its documentation, may imply using the back button to go back to this post sign up interstitial may not work by default.

I agree. the `TaskStackBuilder` API is normally leveraged when an entirely new back stack needs to be created. It is ideal for a scenario like this where the user has been logged in therefore the previous "logged out" back stack history is no longer applicable. 

Therefore, I would conclude that the back button is working as expected, as the user is not to be taken back to the Based on my observation it would seem that the the back button does not take the user back to the "post-signup interstitial" because that user flow was already completed. So if the user wants to add create a new site or add an existing one then they should utilize My Site, as that is now the default flow based on how our signed-in experience is designed.

Let me know if there are alternative thoughts as I would be happy to dig into this a bit further.

**Comment by dcalhoun** [comment]
> I would conclude that the back button is working as expected

From a code perspective, I would agree. From a UX perspective, there may be room to interpret the outcome as unexpected, which is obviously the origin of this issue. 

Rather than altering the use of `TaskStackBuilder` and the back button behavior, an alternative could be updating the design of the My Site tab when an account has no sites to display the two options from the onboarding dialog — "Create a WordPress.com site" and "Add self-hosted site,"  instead of merely "Add new site." 

| My Site Tab | Add New Site Dialog | My Site Tab w/ Two Options |
| - | - | - |
| ![Empty My Site tab includes "Add new site button"](https://user-images.githubusercontent.com/438664/141839489-3a21b505-ede5-4d19-8fa9-6c288af11047.jpg)  | ![Tapping "Add new site button" displays the same two options from the onboarding dialog](https://user-images.githubusercontent.com/438664/141839513-19dc25f8-fb28-4805-ac1f-c791cc01203b.jpg) |  ![The two onboarding options could be placed directly on the My Site tab](https://user-images.githubusercontent.com/438664/141840289-1bf0d096-0cb5-4074-9043-48e8ff1c4622.jpg) |

That said, I might posit having an "Add new site" button that reveals the two options found in the onboarding dialog to be a fairly expected experience. It also reduces the number of call to actions for new users. I.e. we may be safe to close this issue without further changes to the app. 

WDYT?

**Comment by jd-alexander** [comment]
> From a code perspective, I would agree. From a UX perspective, there may be room to interpret the outcome as unexpected, which is obviously the origin of this issue.

Agreed.

> Rather than altering the use of TaskStackBuilder and the back button behavior, an alternative could be updating the design of the My Site tab when an account has no sites to display the two options from the onboarding dialog — "Create a WordPress.com site" and "Add self-hosted site," instead of merely "Add new site."

I am more fond of this idea. I like that the changes would mimic the two options as you said. A designer would need weigh in on the flow being proposed to verify and implement the new design.

> That said, I might posit having an "Add new site" button that reveals the two options found in the onboarding dialog to be a fairly expected experience. It also reduces the number of call to actions for new users. I.e. we may be safe to close this issue without further changes to the app.

> WDYT?

I agree that we could close this issue for now because the "Add new site" button shows the same options therefore the user's inability to get back to that screen doesn't sacrifice much from an UI/UX standpoint. ( this is just a hunch though as we would need to run an A/B test of some sorts to get the facts.)

**Comment by thehenrybyrd** [comment]
I ran into this again while smoke testing, and it still surprises me. As @dcalhoun mentions:

> From a UX perspective, there may be room to interpret the outcome as unexpected, which is obviously the origin of this issue.

When I press a back button, I do expect it to show the screen I was just on. So finding myself on a different screen is suprising.

> display the two options from the onboarding dialog — "Create a WordPress.com site" and "Add self-hosted site,"

This could remove some of the disorientation - especially if they were presented in a similar manner to the onboarding dialog.

**Comment by jkmassel** [comment]
This issue is no longer applicable now that we've adopted the WP.com OAuth flow.
